app/src/main/kotlin/com/hippo/ehviewer/ui/screen/GalleryCommentsScreen.kt

```diff
@@ -108,41 +111,69 @@ import eu.kanade.tachiyomi.util.lang.launchIO
 import eu.kanade.tachiyomi.util.lang.withUIContext
 import eu.kanade.tachiyomi.util.system.logcat
 import kotlin.math.roundToInt
+import kotlin.sequences.forEach
 import kotlinx.coroutines.launch
 import moe.tarsin.coroutines.runSuspendCatching
-import rikka.core.res.resolveColor
 
-private fun Context.generateComment(
-    textView: TextView,
-    comment: GalleryComment,
-): CharSequence {
-    val sp = comment.comment.parseAsHtml(imageGetter = CoilImageGetter(textView))
-    val ssb = SpannableStringBuilder(sp)
-    if (0L != comment.id && 0 != comment.score) {
-        val score = comment.score
-        val scoreString = if (score > 0) "+$score" else score.toString()
-        ssb.append("  ").inSpans(
-            RelativeSizeSpan(0.8f),
-            StyleSpan(Typeface.BOLD),
-            ForegroundColorSpan(theme.resolveColor(android.R.attr.textColorSecondary)),
-        ) {
-            append(scoreString)
+private const val IMAGE_OBJ = '￼'
+private val IMAGE_PATTERN = Regex("<img src=\"([^\"]+)\"")
+private val URL_PATTERN = Regex("(http|https)://[a-z0-9A-Z%-]+(\\.[a-z0-9A-Z%-]+)+(:\\d{1,5})?(/[a-zA-Z0-9-_~:#@!&',;=%/*.?+$\\[\\]()]+)?/?")
+
+fun breakToTextAndUrl(origin: String, text: AnnotatedString): TextOrUrlList {
+    val urls = IMAGE_PATTERN.findAll(origin).toList()
+    if (urls.isEmpty()) return listOf(text.right())
+    val iter = urls.iterator()
+    var currentOfs = 0
+    return buildList<TextOrUrl> {
+        while (true) {
+            val index = text.text.indexOf(IMAGE_OBJ, currentOfs)
+            if (index == -1) break
+            add(text.subSequence(currentOfs, index).right())
+            add(iter.next().groupValues[1].left())
+            currentOfs = index + 1
         }
+        add(text.subSequence(currentOfs, text.length).right())
     }
-    if (comment.lastEdited != 0L) {
-        val str = getString(
-            R.string.last_edited,
-            ReadableTime.getTimeAgo(comment.lastEdited),
-        )
-        ssb.append("\n\n").inSpans(
-            RelativeSizeSpan(0.8f),
-            StyleSpan(Typeface.BOLD),
-            ForegroundColorSpan(theme.resolveColor(android.R.attr.textColorSecondary)),
-        ) {
-            append(str)
+}
+
+@Composable
+fun processComment(
+    comment: GalleryComment,
+    linkStyle: TextLinkStyles,
+    onLinkClick: LinkInteractionListener,
+) = AnnotatedString.fromHtml(comment.comment, linkStyle, onLinkClick).let { text ->
+    buildAnnotatedString {
+        val onSurfaceVariant = MaterialTheme.colorScheme.onSurfaceVariant
+        append(text)
+        URL_PATTERN.findAll(text).forEach { result ->
+            val start = result.range.first
+            val end = result.range.last + 1
+            if (!text.hasLinkAnnotations(start, end)) {
+                addLink(LinkAnnotation.Url(result.groupValues[0], linkStyle, onLinkClick), start, end)
+                addStyle(SpanStyle(textDecoration = TextDecoration.Underline), start, end)
```

Comment:
If there's strike-through on the text, this will override that.

Reply:
Done